e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/T_TwinBrowseOrchestratedTestTheory.cs

```diff
@@ -0,0 +1,175 @@
+            var testPlc = simulatedOpcServer.Values.First();
+
+            TestHelper.Registry_RegisterServerAsync(_context, testPlc.EndpointUrl, cts.Token).GetAwaiter().GetResult();
+
+            _context.ConsumedOpcUaNodes[testPlc.EndpointUrl] = _context.GetEntryModelWithoutNodes(testPlc);
+            dynamic json = TestHelper.WaitForDiscoveryToBeCompletedAsync(_context, cts.Token, new List<string> { testPlc.EndpointUrl }).GetAwaiter().GetResult();
+
+            var numberOfItems = (int)json.items.Count;
+            bool found = false;
+
+            for (int indexOfTestPlc = 0; indexOfTestPlc < numberOfItems; indexOfTestPlc++) {
+
+                var endpoint = ((string)json.items[indexOfTestPlc].discoveryUrls[0]).TrimEnd('/');
+                if (TestHelper.IsUrlStringsEqual(endpoint, testPlc.EndpointUrl)) {
+                    found = true;
+                    break;
+                }
+            }
+
+            Assert.True(found, "OPC Application not activated");
+
+            if (string.IsNullOrWhiteSpace(_context.OpcUaEndpointId)) {
+                WaitForEndpointDiscoveryAndStoreInContext();
+                Assert.False(string.IsNullOrWhiteSpace(_context.OpcUaEndpointId), "The endpoint was not set");
+            }
+
+            TestHelper.Registry_ActivateEndpointAsync(_context, cts.Token).GetAwaiter().GetResult();
+        }
+
+        [Fact, PriorityOrder(2)]
+        public void D1_Discover_OPC_UA_Endpoints() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var endpoints = TestHelper.Registry_GetEndpointsAsync(_context, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.SingleOrDefault(e => string.Equals(_context.OpcUaEndpointId, e.Id));
+
+            Assert.False(endpoint.Id == null, "The endpoint was not found");
+            Assert.Equal(TestConstants.StateConstants.ActivatedAndConnected, endpoint.ActivationState);
+            Assert.Equal(TestConstants.StateConstants.Ready, endpoint.EndpointState);
+        }
+
+        [Fact, PriorityOrder(3)]
+        public void T1_Browse_Address_Space() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var nodes = TestHelper.Twin_GetBrowseEndpointAsync(_context, _context.OpcUaEndpointId, null, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+        }
+
+        [Fact, PriorityOrder(4)]
+        public void T1_1_BrowseSpecificNode() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var nodes = TestHelper.Twin_GetBrowseEndpointAsync(_context, _context.OpcUaEndpointId, null, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            var nodeId = nodes.First().NodeId;
```

Comment:
we should specifically test the expected nodeId. This operation should be deterministic and can be testsd